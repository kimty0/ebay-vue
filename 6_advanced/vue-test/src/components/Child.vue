<template>
  <div>
    <button @click="testEvent">test event</button>
  </div>
</template>

<script>
export default {
  methods: {
    testEvent() {
      this.$emit('test:event');
    }
  }
}
</script>