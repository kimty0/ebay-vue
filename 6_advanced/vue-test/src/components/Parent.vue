<template>
  <div>
    <p>{{ message }}</p>
    <child @test:event="message = 'a message from Child'"></child>
  </div>
</template>

<script>
import Child from './Child.vue';

export default {
  components: {
    Child
  },
  data() {
    return {
      message: 'a message from Parent'
    }
  }
}
</script>