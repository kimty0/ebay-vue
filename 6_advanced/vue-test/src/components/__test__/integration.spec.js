import { mount } from '@vue/test-utils'
import Parent from '../Parent.vue'
import Child from '../Child.vue'

describe('Component Integration Test', () => {
  let wrapper, childComponent, parentComponent;
  beforeEach(() => {
    wrapper = mount(Parent);
    childComponent = wrapper.find(Child);
    parentComponent = wrapper.find(Parent);
  });

  test('[Event Emit] emitting event changes the message', () => {
    // Trigger click event in Child Component
    const testButton = childComponent.find('button');
    testButton.trigger('click');

    // Check the data value if it has been modified by the event
    expect(wrapper.vm.message).toBe('a message from Child');
  });

  test('[Props Change] emitting event changes the rendered output', () => {
    // Trigger click event in Child Component
    const testButton = childComponent.find('button');
    testButton.trigger('click');

    // match the rendered output with changed data
    const pTag = parentComponent.find('p');
    expect(pTag.html()).toBe('<p>a message from Child</p>');
  });

  test('[Rendered Output] snapshot test', () => {
    // Trigger click event in Child Component
    const testButton = childComponent.find('button');
    testButton.trigger('click');
    
    // For log
    expect(parentComponent).toMatchSnapshot();
  })
});