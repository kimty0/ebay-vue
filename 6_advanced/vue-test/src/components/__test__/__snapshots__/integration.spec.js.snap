// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Integration Test [Rendered Output] snapshot test 1`] = `
<div>
  <p>a message from Child</p>
  <div><button>test event</button></div>
</div>
`;
